game/src/contrib/hud/Button.java

```diff
@@ -0,0 +1,122 @@
+package contrib.hud;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.graphics.Texture;
+import com.badlogic.gdx.graphics.g2d.Batch;
+import com.badlogic.gdx.scenes.scene2d.InputEvent;
+import com.badlogic.gdx.scenes.scene2d.InputListener;
+
+import core.Game;
+import core.utils.components.draw.TextureMap;
+
+import java.util.function.Consumer;
+
+public class Button {
+
+    private static final Texture TEXTURE_BUTTON;
+    private static final Texture TEXTURE_BUTTON_HOVER;
+    private static final Texture TEXTURE_BUTTON_PRESS;
+
+    static {
+        TEXTURE_BUTTON = TextureMap.instance().textureAt("hud/button/button_idle.png");
+        TEXTURE_BUTTON_HOVER = TextureMap.instance().textureAt("hud/button/button_hover.png");
+        TEXTURE_BUTTON_PRESS = TextureMap.instance().textureAt("hud/button/button_press.png");
+    }
+
+    protected CombinableGUI parent;
+    protected int x, y, width, height;
+    private boolean pressed = false;
+    private Consumer<Button> onClick;
+
+    public Button(CombinableGUI parent, int x, int y, int width, int height) {
+        this.x = x;
+        this.y = y;
+        this.width = width;
+        this.height = height;
+        this.parent = parent;
+        this.init();
+    }
+
+    private void init() {
+        this.parent
+                .actor()
+                .addListener(
+                        new InputListener() {
+                            @Override
+                            public boolean touchDown(
```

Comment:
hier wäre ein kommentar ganz hilfreich: Was macht das? z.B `//add drag and drop`